doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,375 @@
+# The Node.js Event Loop, Timers, and `process.nextTick()`
+
+## What is the Event Loop?
+
+The event loop is what allows Node.js to perform non-blocking I/O
+operations — despite the fact that JavaScript is single-threaded — by
+offloading operations to the system kernel whenever possible.  
+
+Since most modern kernels are multi-threaded, they can handle
+multiple operations executing in the background.  When one of these
+operations completes, the kernel tells Node.js so that the appropriate callback 
+may added to the `poll` queue to eventually be executed.  We'll explain
+this in further detail later in this topic.  
+
+## Event Loop Explained 
+
+When Node.js starts, it initializes the event loop, processes the provided
+input script (or drops into the REPL, which is not covered in this document)
+which may make async API calls, schedule timers, or call `process.nextTick()`, 
+then begins processing the event loop.  
+
+The Following diagram shows a simplified overview of the event loop's
+order of operations.
+
+       ┌───────────────────────┐
+    ┌─>│        timers         │
+    │  └──────────┬────────────┘
+    │  ┌──────────┴────────────┐
+    │  │   pending callbacks   │
+    │  └──────────┬────────────┘      ┌───────────────┐
+    │  ┌──────────┴────────────┐      │   incoming:   │
+    │  │         poll          │<─────│  connections, │
+    │  └──────────┬────────────┘      │   data, etc.  │
+    │  ┌──────────┴────────────┐      └───────────────┘
+    └──│     setImmediate      │
+       └───────────────────────┘
+
+*note: each box will be referred to as a "phase" of the event loop.*
+
+Each phase has a FIFO queue of callbacks to execute. While each phase  is
+special in its own way, generally, when the event loop enters a  given phase,
+it will perform any operations specific to that phase,  then execute callbacks
+in that phase's queue until the queue has been exhausted or the maximum number
+of callbacks have executed.  When the queue has been exhausted or the callback
+limit is reached, the event loop will move to the next  phase, and so on.
```

Comment:
Looks like there are many of these mid-sentence double spaces, please clean them up :wink: